site/sfguides/src/build_a_private_custom_api_in_python/build_a_private_custom_api_in_python.md

```diff
@@ -0,0 +1,463 @@
+
+```sql
+CALL SYSTEM$GET_SERVICE_STATUS('API_OCKAM_OUTLET');
+CALL SYSTEM$GET_SERVICE_LOGS('API_OCKAM_OUTLET', '0', 'ockam-outlet', 1000);
+```
+
+#### Create an Ockam node next to the API Client
+
+Finally, create an Ockam node in the environment where your API client is running.
+
+
+Run the following command in the codespace terminal:
+
+```sh
+docker run --rm -d  --name ockam-inlet -p 8001:8001 \
+  ghcr.io/build-trust/ockam node create --foreground \
+  --enrollment-ticket "$(ockam project ticket --usage-count 1 --expires-in 1h --attribute snowflake-api-service-inlet)" \
+  --configuration "
+    tcp-inlet:
+      from: 0.0.0.0:8001
+      via: snowflake-api-service-relay
+      allow: snowflake-api-service-outlet
+  "
+```
+
+### Testing using cURL
+
+The API can now be tested using the cURL command-line tool from a new codespace terminal.
+
+#### Top 10 Customers
+
+To retrieve the top 10 customers in the date range of `1995-02-01` to `1995-02-14` using the Snowflake Connector for Python, run:
+
+```bash
+curl -X GET "http://localhost:8001/connector/customers/top10?start_range=1995-02-01&end_range=1995-02-14"
+```
+
+To retrieve the top 10 customers in the date range of `1995-02-01` to `1995-02-14` using the Snowflake Snowpark API, run:
+
+```bash
+curl -X GET "http://localhost:8001/snowpark/customers/top10?start_range=1995-02-01&end_range=1995-02-14"
+```
+
+If you call the endpoint without specifying the `start_range` then `1995-01-01` will be used. If you call the endpoint without specifying the `end_range` then `1995-03-31` will be used.
+
+#### Monthly sales for a given year for a sales clerk
+To retrieve the monthly sales for clerk `000000002` for the year `1995` using the Snowflake Connector for Python, run:
+
+```bash
+curl -X GET "http://localhost:8001/connector/clerk/000000002/yearly_sales/1995"
+```
+
+To retrieve the monthly sales for clerk `000000002` for the year `1995` using the Snowflake Snowpark API, run:
+
+```bash
+curl -X GET "http://localhost:8001/connector/clerk/000000002/yearly_sales/1995"
+```
+
+<!-- ------------------------ -->
+## Stopping the API
+Duration: 1
+
+To stop the API, you can suspend the service. From the Snowflake console or SnowSQL, run:
+
+```sql
+USE ROLE DATA_API_ROLE;
+ALTER SERVICE API.PRIVATE.API SUSPEND;
+```
+
+<!-- ------------------------ -->
+## Cleanup
+Duration: 2
+
+To fully remove everything you did today you only need to drop some objects in your Snowflake account. From the Snowflake console or SnowSQL, as `DATA_API_ROLE` run:
+
+```SQL
+USE ROLE DATA_API_ROLE;
+
+DROP SERVICE API.PRIVATE.API;
+DROP SERVICE API_OCKAM_OUTLET;
+```
+
+Then, as `ACCOUNTADMIN` run:
+
+```SQL
+USE ROLE ACCOUNTADMIN;
+
+DROP ROLE IF EXISTS DATA_API_ROLE;
+DROP DATABASE IF EXISTS API;
+DROP INTEGRATION IF EXISTS OCKAM;
+DROP COMPUTE POOL IF EXISTS API_POOL;
+DROP WAREHOUSE IF EXISTS DATA_API_WH;
+```
+
+<!-- ------------------------ -->
+## Conclusion
```

Comment:
@mrinalwadhwa Pl rename this step to **Conclusion And Resources** and also provide links to relevant resources. For example, https://quickstarts.snowflake.com/guide/getting_started_with_dataengineering_ml_using_snowpark_python/index.html#7

Reply:
@iamontheinet great suggestion.
Just pushed an updated version with a `Conclusion and Resources` section https://github.com/Snowflake-Labs/sfquickstarts/pull/1521/commits/d9c0eff225ec8d66c2b9738218273f8291309208